Milestone : Modal pop up

AngomRanjan opened the pull request:
## Profile - Milestone : Modal pop up

#### In this milestone :

- Created A JS Object
- Created An Array of Object To Store Project Data.
- Created A Modal Pop-up Window To Display Project Details.

AngomRanjan committed:
Add Function To Store Project Info

AngomRanjan committed:
Add Click Event Handler to Project Buttons

AngomRanjan committed:
Test run Show Detail in Alert Box

AngomRanjan committed:
Add Modal Pop-UP UI

AngomRanjan committed:
Link JS object to Modal Pop-Up

AngomRanjan committed:
Style Mobile Version Pop-Up

AngomRanjan committed:
Style Desktop Modal Pop Up

AngomRanjan committed:
Add Links To Modal Button

Ismailco reviewed (commented):
Hi @AngomRanjan,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that need to be created using peer programming, but not everybody in the pair-programming group has contributed code on this project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

gasangw committed:
update styles.css file

Remove reductant work.

AngomRanjan commented:
> 

my coding partner made the contribution

thecodechaser commented on index.html:
```diff
@@ -95,7 +95,7 @@ <h2 class="project-titles">Tonic</h2>
               <li class="tags">javascripts</li>
             </ul>
-            <button class="btn" type="button">See Project</button>
+            <button class="btn btn-projects" type="button" id="1">See Project</button>
```
- [ ] You need to update the main page so that the projects section is created dynamically using the information stored in that JavaScript object. Remember, all of the HTML in that section is created when the page loads. (Please make sure the entire project section is created using javascript dynamically, You may want to remove the project section from the HTML file and create it using javascript).

danmainah replied:
- [x] Kindly ensure all your projects are loaded dynamically, currently, you have hardcoded all the projects in your HTML file. kindly create an array in your profile.js file that contains all the projects.

thecodechaser commented on profile.js:
```diff
@@ -35,4 +35,64 @@ document.getElementById('menu').addEventListener('click', () => {
\ No newline at end of file
+});
+// || =========== Pop-Up Modal ============ ||
+const arrProjects = [];
+function addProjects() {
+  // Stores Project Infos in arrProject
+  const colTitles = document.getElementsByClassName('project-titles');
+  const colClient = document.getElementsByClassName('client');
+  const colRole = document.getElementsByClassName('type');
+  const colYear = document.getElementsByClassName('year');
+  for (let i = 0; i < colTitles.length; i += 1) {
+    // | creating a new project object an pushing it to arrProject
+    const objProject = {};
+    objProject.name = colTitles[i].textContent;
+    objProject.client = colClient[i].textContent;
+    objProject.role = colRole[i].textContent;
+    objProject.year = colYear[i].textContent;
+    objProject.tags = ['html', 'css', 'javascript', 'github', 'ruby', 'bootstrap'];
+    objProject.snapshot = `images/SnapshootD${i + 1}.jpg`;
+    arrProjects.push(objProject);
+  }
+}
```
- [ ] Please make sure you store all project information in an array of objects, Right now you are getting the Infos from the HTML file and storing it in your objects, But you should store all the infos manually in objects so you can easily create project section and pop-up model.

thecodechaser commented on index.html:
```diff
@@ -237,6 +238,49 @@ <h2 class="head-1" id="contact-me">Contact Me</h2>
+    <div class="modal-ui md-hide" id="modal-ui-wrapper">
+      <div class="modal-card" id="modal-card">
+        <div class="mc-titlebar" id="mctitlebar">
+          <h2 class="mc-title" id="mc-title">Tonic</h2>
+          <button type="button" class="mc-btn" id="mc-close"> &times;</button>
+        </div>
+        <ul class="inline-li project-summary">
+          <li class="client" id="mc-client">CANOPY</li>
+          <li class="type" id="mc-type">Back End Dev</li>
+          <li class="year" id="mc-year">2015</li>
+        </ul>
+        <div class="mc-snap-place">
+          <img class="mc-snapshoot" src="images/SnapshootD1.jpg" alt="Snapshot of Project Tonic" id="mc-img">
+        </div>
+        <div class="mc-l-block">
+          <div class="detail mc-l-block-l" id='mc-details'>
+            <p class="project-descriptions" id="mc-desc">
+              Lorem Ipsum is simply dummy text of the printing and typesetting industry. 
+              Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, 
+              when an unknown printer took a galley of type and scrambled it to make a 
+              type specimen book. It has survived not only five centuries, but also the leap 
+              into electronic typesetting, remaining essent
+            </p>
+          </div>
+          <div class="mc-l-block-r">
+            <ul class="inline-li">
+              <li class="tags mc-tags">html</li>
+              <li class="tags mc-tags">css</li>
+              <li class="tags mc-tags">javascripts</li>
+              <li class="tags mc-tags">github</li>
+              <li class="tags mc-tags">ruby</li>
+            </ul>
+            <div class="modal-btn-area">
+              <button class="btn btn-modal" type="button" onclick="location.href='https://angomranjan.github.io/Profile/'">See Live <img src="icons/Icon.png" alt="" class="btn-icn"></button>
+              <button class="btn btn-modal" type="button" onclick="location.href='https://github.com/AngomRanjan/Profile'">See Source <img src="icons/Vector.png" alt="" class="btn-icn"></button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>    
```
- [ ] Please make sure the model is created using javascript dynamically, Right now you have created the model in HTML and replaced its content using javascript but it should be created dynamically.

thecodechaser commented on profile.js:
```diff
@@ -35,4 +35,64 @@ document.getElementById('menu').addEventListener('click', () => {
+  return arrProjects[proID];
+}
+
+function showProjectDetails(proID) {
+  // showing  Project Info
+  const modPopUp = document.getElementById('modal-ui-wrapper');
+  const p = getProjectDetails(proID - 1);
+  document.getElementById('mc-title').textContent = p.name;
+  document.getElementById('mc-client').textContent = p.client;
```
- [x] You did a very good job on the styles but for this project, you are supposed to stick to the [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=34%3A1143) design as much as possible. Please increase the width of the model and top-bottom margins and make sure it matches the template.

### Your design

![image](https://user-images.githubusercontent.com/83803414/169248295-c1bb2c86-eb23-4916-ab85-8e6ebe513273.png)

### Figma design

![image](https://user-images.githubusercontent.com/83803414/169248204-0b257ca1-1c6f-47f5-b5a0-8c89a5ce5e65.png)

thecodechaser replied:
- [x] Please use the correct font-weight for the project description and make sure it matches the Template.

### Your design

![image](https://user-images.githubusercontent.com/83803414/169248974-59a33c15-a87c-473c-8af3-f58f10921030.png)

### Figma design

![image](https://user-images.githubusercontent.com/83803414/169248934-338784a8-3ae1-4da2-afd8-60eb102dbd2e.png)

thecodechaser commented on profile.js:
```diff
@@ -35,4 +35,64 @@ document.getElementById('menu').addEventListener('click', () => {
+  return arrProjects[proID];
+}
+
+function showProjectDetails(proID) {
+  // showing  Project Info
+  const modPopUp = document.getElementById('modal-ui-wrapper');
```
- [ ] Please use the correct width for model buttons and make sure it matches the template.

### Your design

![image](https://user-images.githubusercontent.com/83803414/169249283-de3fc65a-0055-4444-86a4-cbd7de4135ec.png)

### Figma design

![image](https://user-images.githubusercontent.com/83803414/169249467-8f19af11-dbb0-43fc-8c39-0e8f047a21df.png)

thecodechaser commented on profile.js:
```diff
@@ -35,4 +35,64 @@ document.getElementById('menu').addEventListener('click', () => {
+  return arrProjects[proID];
+}
+
+function showProjectDetails(proID) {
+  // showing  Project Info
```
- [x] For mobile screens please fix the model buttons' width and make sure it matches the template.

### Your design

![image](https://user-images.githubusercontent.com/83803414/169249876-eccfc548-2d09-42f3-9be0-6af692dd41f5.png)

### Figma design

![image](https://user-images.githubusercontent.com/83803414/169249920-d4af0878-3503-494e-b525-a8566c200588.png)

AngomRanjan replied:
Made All the Changes as suggested by you.

thecodechaser reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @AngomRanjan, and @gasangw👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AngomRanjan committed:
Add Code To Generate Modal PopUp Dynamically

AngomRanjan committed:
Made Changes as suggested

AngomRanjan commented:
Made Changes As Suggested

danmainah commented on profile.js:
```diff
@@ -35,4 +35,113 @@ document.getElementById('menu').addEventListener('click', () => {
+  return itemDetails;
+}
+
+function addProjects() {
+  // Stores Project Infos in arrProject
+  const colTitles = ['Tonic', 'Multi-Post Stories', 'Facebook 360', 'Uber Navigation'];
+  const colClient = ['Canopy', 'Facebook', 'Facebook', 'Uber'];
+  const colRole = ['Back End', 'Full Stack', 'Full Stack', 'Lead Role'];
+  const colYear = ['2015', '2015', '2015', '2018'];
+  for (let i = 0; i < 4; i += 1) {
+    // | creating a new project object an pushing it to arrProject
+    const objProject = {};
+    objProject.name = addTitle(colTitles[i]);
+    objProject.summary = addSummary([colClient[i], colRole[i], colYear[i]]);
+    objProject.snapshot = addSnapshot();
+    objProject.details = addDetails();
+    arrProjects.push(objProject);
+  }
+}
```
- [x] Kindy makes sure you hardcode all your projects in the array not pushing into it the projects as you have done. This will ensure the easy addition of projects to your portfolio in the future.

danmainah commented on profile.js:
```diff
@@ -35,4 +35,113 @@ document.getElementById('menu').addEventListener('click', () => {
\ No newline at end of file
+});
+// || =========== Pop-Up Modal ============ ||*/
+const arrProjects = [];
```
- [x] [optional] Kindly check the design of your project at different widths it changes breaking the design.

danmainah reviewed (changes requested):
Hi @AngomRanjan and @gasangw , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design
✔️  Good PR and Description


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AngomRanjan committed:
Recreate Works Section Dynamically Using JS

- Remove Works Section (project) from main html
- Recreate Works Dynamically on page load Using js
- Create A Dynamic Modal PopUp with a Functional Close button

AngomRanjan committed:
Changes Made as Suggested

AngomRanjan commented:
#### Made Changes as suggested
- Remove Works Section From Main HTML as Suggested
- Recreate Works Section Dynamically on Page Load Using Js array/ object
- Create Modal PopUp Dynamically

smunozmo reviewed (approved):
Hi @AngomRanjan, 👋 

## Highlights 👏 

- Content is generated dynamically, perfect!
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

### APPROVED! 💯  💯  💯 

<img src="https://media4.giphy.com/media/cFkiFMDg3iFoI/giphy.gif"/>


Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
